bindata/network/network-metrics/002-daemonset.yaml

```diff
@@ -0,0 +1,74 @@
+---
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: network-metrics-daemon
+  namespace: openshift-network-metrics
+  annotations:
+    kubernetes.io/description: |
+      This daemonset launches the network metrics daemon on each node
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+    networkoperator.openshift.io/non-critical: ""
+spec:
+  selector:
+    matchLabels:
+      app: network-metrics-daemon
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: network-metrics-daemon
+        component: network
+        type: infra
+        openshift.io/component: network
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/worker: ""
+      tolerations:
+        - operator: Exists
+      containers:
+        - name: network-metrics-daemon
+          image: {{.NetworkMetricsImage}}
+          command:
+            - /usr/bin/network-metrics
+          resources:
+            requests:
+              cpu: 10m
+              memory: 100Mi
+          imagePullPolicy: Always
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
```

Comment:
These get read by the network-metrics-daemon process? Stylistically, I think it's better to pass them as command-line flags, rather than having its configuration be a mix of command-line flags and environment variables.

Reply:
The namespace env var is not read any more and I can probably remove it.
For the other one I agree, but I could not find a way to pass this kind of information as an argument. Using the downward api to expose pod fields as env variable seem to be the suggested approach.

Are you aware of an alternative approach to pass a pod field as an argument of the container?

Reply:
Yeah, you have to use the downward API to set the environment variable but then you just use the variable from the command-line rather than from the code. eg,

    args:
      ...
      - --node-name "${NODE_NAME}"

Reply:
Ah, super simple! Thanks.